lib/internal/bootstrap/node.js

```diff
@@ -340,6 +340,10 @@ function startup() {
     NativeModule.require('internal/process/report').setup();
   }
 
+  if (getOptionValue('--frozen-intrinsics')) {
```

Comment:
I'd say `prepareUserCodeExecution` is a better place for this, as this does not make a lot of sense in use cases like `node --help`

Reply:
Thanks. This is now in `pre_execution.js`. Let me know if that seems right.